### PR TITLE
Bump module to v4

### PR DESCRIPTION
**Pull request by adambarreiro**
This PR bumps the provider module to v4 and starts consuming v3 from https://github.com/vmware/go-vcloud-director

Unit tests passed.

### CODE REVIEW

File: .changes/1344-notes.md [hunk 1]
```diff
@@ -0,0 +1 @@
+* Bump module version from `v3` to `v4`: `github.com/vmware/terraform-provider-vcd/v4` [GH-1344]
```
Comment (Didainius) [inline .changes/1344-notes.md:1]: We also need to modify 
* `.goreleaser.yaml` in line 15
* `GNUmakefile`

File: go.mod [hunk 2]
```diff
@@ -7,7 +7,9 @@ require (
 	github.com/hashicorp/go-version v1.6.0
 	github.com/hashicorp/terraform-plugin-sdk/v2 v2.34.0
 	github.com/kr/pretty v0.3.1
-	github.com/vmware/go-vcloud-director/v2 v2.26.0
+	github.com/vmware/go-vcloud-director/v3 v3.0.0-alpha.2
+	// TODO: TM: Change this to v4
+	github.com/vmware/terraform-provider-vcd/v3 v3.14.0
```
Comment (Didainius) [inline go.mod:12]: Updating `main.go` to use `v4` should get rid of this import alltogether

File: vcd/provider.go [hunk 3]
```diff
@@ -6,13 +6,13 @@ import (
...
 
 	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
 
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
-	"github.com/vmware/go-vcloud-director/v2/util"
+	"github.com/vmware/go-vcloud-director/v3/util"
 )
 
 // BuildVersion holds version which is meant to be injected at build time using ldflags
```
Comment (Didainius) [inline vcd/provider.go:18]: The comment just below this line (github doesn't allow commenting it) could also be updated

**Review by Didainius — CHANGES_REQUESTED** [review]
A few more changes required

File: go.mod (hunk 2 above)
Comment (adambarreiro) [reply go.mod]: True, fixed

File: vcd/provider.go (hunk 3 above)
Comment (adambarreiro) [reply vcd/provider.go:18]: Fixed

**Review by Didainius — APPROVED** [review]

**Review by lvirbalas — APPROVED** [review]
Important step.

**Pull request merged**